packages/commands/src/solana/deposit.ts

```diff
@@ -0,0 +1,166 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Wallet } from "@coral-xyz/anchor";
+import {
+  AccountLayout,
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { clusterApiUrl, PublicKey } from "@solana/web3.js";
+import GATEWAY_DEV_IDL from "@zetachain/protocol-contracts-solana/dev/idl/gateway.json";
+import GATEWAY_PROD_IDL from "@zetachain/protocol-contracts-solana/prod/idl/gateway.json";
+import * as bip39 from "bip39";
+import bs58 from "bs58";
+import { Command, Option } from "commander";
+import { ethers } from "ethers";
+
+const networks = ["devnet", "localnet", "mainnet"];
+
+export interface DepositOptions {
+  amount: string;
+  from: string;
+  mint: string;
+  mnemonic: string;
+  network: string;
+  privateKey: string;
+  recipient: string;
+  to: string;
+  tokenProgram: string;
+}
+
+export const keypairFromMnemonic = async (
+  mnemonic: string
+): Promise<anchor.web3.Keypair> => {
+  const seed = await bip39.mnemonicToSeed(mnemonic);
+  const seedSlice = new Uint8Array(seed).slice(0, 32);
+  return anchor.web3.Keypair.fromSeed(seedSlice);
+};
+
+export const keypairFromPrivateKey = (
+  privateKey: string
+): anchor.web3.Keypair => {
+  try {
+    const decodedKey = bs58.decode(privateKey);
+    return anchor.web3.Keypair.fromSecretKey(decodedKey);
+  } catch (error) {
+    throw new Error(
+      "Invalid private key format. Expected base58-encoded private key."
+    );
+  }
+};
+
+const main = async (options: DepositOptions) => {
+  // Mainnet and devnet use the same IDL
+  const gatewayIDL =
+    options.network === "localnet" ? GATEWAY_DEV_IDL : GATEWAY_PROD_IDL;
+
+  let keypair: anchor.web3.Keypair;
+  if (options.privateKey) {
+    keypair = keypairFromPrivateKey(options.privateKey);
+  } else if (options.mnemonic) {
+    keypair = await keypairFromMnemonic(options.mnemonic);
+  } else {
+    throw new Error("Either privateKey or mnemonic must be provided");
```

Comment:
This shouldn't be necessary if we validate params with a solid Zod schema.

Reply:
Right now we're duplicating logic: optionality, default values, conflicts are handled both in commander and zod.

Reply:
I can see advantages of zod for exported functions, but less so for commands that already use declarative syntax to do validation.

For example,

```ts
  .refine((data) => !(data.mnemonic && data.privateKey), {
    message: "Only one of mnemonic or privateKey can be provided, not both",
  });
```

And:

```ts
    .addOption(
      new Option("--name <name>", "Account name")
        .conflicts(["private-key"])
    )
    .addOption(
      new Option("--private-key <key>", "Private key for signing transactions")
        .conflicts(["name"])
    )
```

We don't need both.

Reply:
Maybe we do types in zod like `.string()`, but default values and validation in commander.

Reply:
Default values and enum values should definitely be in commander, because they show up in `--help`.

Reply:
Added zod validation: a4f049f